### PR TITLE
IMPR/GRID-265

### PR DESCRIPTION
**Pull request by Dwaynekj**
@joneit @rdepena 
Paired PR with https://github.com/openfin/fin-hypergrid/pull/323

Admittedly Base could be inherited from fin-hypergrid DataSourceBase.js

### CODE REVIEW

**Comment by joneit** [comment]
@Dwaynekj As we've discussed, DataSourceBase.js can become an npm module which both our projects can then reference. (Any any one else who wants to write their own data transformers.)

**Comment by Dwaynekj** [comment]
We should discuss with @rdepena

File: js/DataSourceGlobalFilter.js [hunk 1]
```diff
@@ -41,6 +41,10 @@ var DataSourceGlobalFilter = DataSourceIndexed.extend('DataSourceGlobalFilter',
     get: function(filter) {
         return this.filter;
     },
+    
+    sortGroups: function(sorter){
+      this.dataSource.sortGroups(sorter);  
+    },
```
Comment (joneit) [inline js/DataSourceGlobalFilter.js]: why is this here? sounds like it needs to be in base.... but wait didn't you state that filter is closer to origin? is this really being called?

File: js/DataSourceAggregator.js [hunk 2]
```diff
@@ -265,6 +265,18 @@ var DataSourceAggregator = Base.extend('DataSourceAggregator', {
...
     },
 
+    /**
+     * @memberOf DataSourceAggregator#
+     * @returns {*|boolean}
+     */
+    isDrillDown: function(event) {
+        var result = this.viewMakesSense();
+        if (result && event) {
+            result = event.dataCell.x === this.treeColumnIndex;
```
Comment (joneit) [inline js/DataSourceAggregator.js]: I don't think "events" should be in here or "dataCell" for that matter. We should just send `columnIndex`.

File: js/Base.js [hunk 3]
```diff
@@ -23,24 +17,104 @@ Base.prototype = {
...
+        }
+    },
+    setHeaders: function(arr) {
+        if (this.dataSource) {
+            return this.dataSource.setHeaders.apply(this.dataSource, arguments);
+        }
+    },
+    getGrandTotals: function(row){
+        //row: Ideally this should be set and get bottom/top totals
+        //Currently this function is just sending the same for both in aggregations
```
Comment (joneit) [inline js/Base.js]: remove comment from here

File: js/Base.js [hunk 4]
```diff
@@ -23,24 +17,104 @@ Base.prototype = {
...
+        if (this.dataSource) {
+            return this.dataSource.click.apply(this.dataSource, arguments);
+        }
+    },
+    isDrillDown: function() {
+        if (this.dataSource) {
+            return this.dataSource.isDrillDown.apply(this.dataSource, arguments);
+        }
+    },
+    apply: function() {},
```
Comment (joneit) [inline js/Base.js]: interesting... so this is a short-stop

File: js/DataSourceAggregator.js (hunk 2 above)
Comment (Dwaynekj) [inline js/DataSourceAggregator.js]: ok. Will fix

File: js/Base.js (hunk 3 above)
Comment (Dwaynekj) [inline js/Base.js]: ok

File: js/Base.js (hunk 4 above)
Comment (Dwaynekj) [inline js/Base.js]: For apply? Yes

File: js/DataSourceGlobalFilter.js (hunk 1 above)
Comment (Dwaynekj) [inline js/DataSourceGlobalFilter.js]: Yea its being called. In the case where the filter takes a group sorter (i.e aggregations and group view)

**Comment by Dwaynekj** [comment]
Changed made
